File: base-images/node-proxy/proxy.js
#!/usr/bin/env node
/**
 * Generic FastMCP stdio->HTTP proxy for Node.js MCP servers
 */
const { spawn } = require('child_process');

// Get configuration from environment
const MCP_PORT = process.env.MCP_PORT || 3000;
const MCP_COMMAND = process.env.MCP_COMMAND || 'node server.js';
const MCP_ARGS = process.env.MCP_ARGS ? process.env.MCP_ARGS.split(' ') : [];

console.log(`Starting Node.js MCP proxy on port ${MCP_PORT}`);
console.log(`Proxying command: ${MCP_COMMAND} ${MCP_ARGS.join(' ')}`);

// This is a simplified version. Full implementation would use
// FastMCP's proxy functionality to forward requests to the stdio process

// Start HTTP server
const http = require('http');
const server = http.createServer((req, res) => {
    res.writeHead(200, {'Content-Type': 'application/json'});
    res.end(JSON.stringify({status: 'Node.js MCP Proxy Running'}));
});

server.listen(MCP_PORT, '0.0.0.0', () => {
    console.log(`Node.js MCP proxy listening on port ${MCP_PORT}`);
});

File: base-images/python-proxy/proxy.py
#!/usr/bin/env python3
"""
Generic FastMCP stdio->HTTP proxy for Python MCP servers
"""
import os
import subprocess
import asyncio
from fastmcp import FastMCP
import uvicorn

# Get configuration from environment
MCP_PORT = int(os.getenv('MCP_PORT', '3000'))
MCP_COMMAND = os.getenv('MCP_COMMAND', 'python server.py')
MCP_ARGS = os.getenv('MCP_ARGS', '').split() if os.getenv('MCP_ARGS') else []

# Create FastMCP app
mcp = FastMCP("Python MCP Proxy")

@mcp.tool()
def proxy_tool():
    """Proxy tool that forwards to the stdio MCP server"""
    pass

def run_mcp_server():
    """Run the actual MCP server as stdio process"""
    cmd = MCP_COMMAND.split() + MCP_ARGS
    return subprocess.Popen(
        cmd,
        stdin=subprocess.PIPE,
        stdout=subprocess.PIPE,
        stderr=subprocess.PIPE,
        text=True
    )

if __name__ == "__main__":
    # Start the FastMCP server with proxy to stdio
    print(f"Starting Python MCP proxy on port {MCP_PORT}")
    print(f"Proxying command: {MCP_COMMAND} {' '.join(MCP_ARGS)}")
    
    # Note: This is a simplified version. Full implementation would use
    # FastMCP's proxy functionality to forward requests to the stdio process
    uvicorn.run(
        "proxy:mcp",
        host="0.0.0.0",
        port=MCP_PORT,
        log_level="info"
    )
